### PR TITLE
Add ref, source and utm_source as referrer

### LINKED ISSUE #169: Referral Tracking
Currently learning Ackee and also using GoatCounter on the side. If I send out a URL:

https://example.com?ref=source

Goat Counter correctly shows the referral came from "source." These do not come up in Ackee, and I sometimes just have "No Referrers" in the past 24 hours if there weren't any other external links to my page.

I currently use this for static links that I send out to see which version that I'm sending is doing better. Kind of like A/B testing, but more manual.

Is this expected behavior within Ackee?

Thanks for making such a minimal and private tracker. I have enjoyed using it so far!

### PR DESCRIPTION
**Pull request by BetaHuhn**
Backend implementation to fix #169

### CODE REVIEW

**Comment by BetaHuhn** [comment]
I am having trouble figuring out how to change the URL scalar to a union in the GraphQL API as I am not ver familiar with this. Also currently the UI displays an error as it expects the referrer id to be a URL not a string,

File: src/aggregations/aggregateNewFields.js [hunk 1]
```diff
@@ -28,7 +28,11 @@ module.exports = (ids, properties, limit) => {
 	]
 
 	properties.forEach((property) => {
-		aggregation[0].$match[property] = { $ne: null }
+		if (or) {
+			(aggregation[0].$match['$or'] = aggregation[0].$match['$or'] || []).push({ [property]: { $ne: null } })
```
Comment (electerious) [inline src/aggregations/aggregateNewFields.js:32]: Smart one, but I had to read it a few times to understand what's going on 😃 Let's split it into two lines.

```js
const osProp = 
aggregation[0].$match['$or'] = aggregation[0].$match['$or'] || []
aggregation[0].$match['$or'].push({ [property]: { $ne: null } })
```

or (but I don't know if that's easier to read)

```js
aggregation[0].$match['$or'] = [...(aggregation[0].$match['$or'] || []), { [property]: { $ne: null } }]
```

File: src/types/records.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ module.exports = gql`
 		"""
 		Where the user came from. Either unknown, a specific page or just the domain. This depends on the browser of the user.
 		"""
-		siteReferrer: URL
+		siteReferrer: String
```
Comment (electerious) [inline src/types/records.js:21]: Please revert this file. I will give unions a try :)

**Review by electerious — CHANGES_REQUESTED** [review]
Thanks 👍 Please let me know when the two comments are resolved. I will merge it into a dedicated branch and add the GraphQL part tomorrow so you can continue with ackee-track if you want.

**Review by electerious — APPROVED** [review]

**Comment by electerious** [comment]
Resolved the issues. The union idea didn't work out. I've adjusted the GraphQL schema to return a String for the referrer id. It now needs a isUrl function in the UI to detect URLs (the currently list only works with URLs).

**Comment by BetaHuhn** [comment]
[electerious/ackee-tracker#27](https://github.com/electerious/ackee-tracker/pull/27) adds support for this feature to the tracker.

**Pull request merged**